### ISSUE TITLE
TLS SNI support for python 2

### ISSUE THREAD

**Issue by kabniel** [xray7224/PyPump#43]
Some pump servers out there (pump.jpope.org) are using TLS [SNI](http://en.wikipedia.org/wiki/Server_Name_Indication) but urllib3 and by extension requests module which pypump depends on doesnt have SNI support by default on Python 2.
This will however be automagically enabled if a [few more modules are installed](https://github.com/kennethreitz/requests/pull/1347).

It looks easy enough if you read the link above, but it's a bit more complicated than just doing `pip install ndg-httpsclient pyopenssl`

On Debian Wheezy i had to `aptitude install python-dev` or else `pyopenssl` will fail to install/build, you also need to install the module `pyasn1` to get SNI requests working.

TL;DR (too late)
`aptitude install python-dev`
`pip install ndg-httpsclient pyopenssl pyasn1`

If possible, it would be great if we could install the modules as requirements for pypump if python 2 is used, but i'm not sure how to deal with the python-dev requirement
